lldb/cmake/modules/LLDBFramework.cmake

```diff
@@ -105,18 +78,37 @@ set_target_properties(liblldb-resource-headers PROPERTIES FOLDER "LLDB/Resources
 add_dependencies(liblldb-resource-headers liblldb-header-staging)
 add_dependencies(liblldb liblldb-resource-headers)
 
+# Glob all necessary header files from source and place them into a list.
+file(GLOB public_headers ${LLDB_SOURCE_DIR}/include/lldb/API/*.h)
+set(generated_public_headers ${LLDB_OBJ_DIR}/include/lldb/API/SBLanguages.h)
+file(GLOB root_public_headers ${LLDB_SOURCE_DIR}/include/lldb/lldb-*.h)
+file(GLOB root_private_headers ${LLDB_SOURCE_DIR}/include/lldb/lldb-private*.h)
+list(REMOVE_ITEM root_public_headers ${root_private_headers})
+set(lldb_framework_header_list ${public_headers} ${generated_public_headers} ${root_public_headers})
+
+set(lldb_framework_header_basename_list)
+foreach(header ${lldb_framework_header_list})
+  get_filename_component(header_basename ${header} NAME)
+  list(APPEND lldb_framework_header_basename_list ${header_basename})
+endforeach()
```

Comment:
You can fold this into the loop below and do the transform on every iteration instead of creating a separate list?